homeassistant/components/system_bridge/sensor.py

```diff
@@ -329,236 +373,231 @@ async def async_setup_entry(
                 translation_key="displays_connected",
                 state_class=SensorStateClass.MEASUREMENT,
                 icon="mdi:monitor",
-                value=lambda _, count=display_count: count,
+                value=lambda data: len(data.displays) if data.displays else None,
             ),
             entry.data[CONF_PORT],
         )
     )
 
-    for _, display in enumerate(displays):
-        entities = [
-            *entities,
-            SystemBridgeSensor(
-                coordinator,
-                SystemBridgeSensorEntityDescription(
-                    key=f"display_{display['name']}_resolution_x",
-                    name=f"Display {display['name']} resolution x",
-                    state_class=SensorStateClass.MEASUREMENT,
-                    native_unit_of_measurement=PIXELS,
-                    icon="mdi:monitor",
-                    value=lambda data, k=display["key"]: getattr(
-                        data.display, f"{k}_resolution_horizontal", None
+    if coordinator.data.displays is not None:
+        for index, display in enumerate(coordinator.data.displays):
+            entities = [
+                *entities,
+                SystemBridgeSensor(
+                    coordinator,
+                    SystemBridgeSensorEntityDescription(
+                        key=f"display_{display.id}_resolution_x",
```

Comment:
Is the display name the same as the display id?

Reply:
No they are different. I should add a note on potential names to the breaking changes